qiskit/providers/aer/jobs/utils.py

```diff
@@ -55,35 +56,139 @@ def wrapper(*args, **kw):
     return wrapper
 
 
-def split_qobj(qobj, max_size=None, qobj_id=None):
-    """Split a qobj and return a list of qobjs each with a single experiment.
+def _copy_qobj_for_noise(qobj, max_shot_size, qobj_id):
 
-    Args:
-        qobj (Qobj): The input qobj object to split
-        max_size (int or None): the maximum number of circuits per job. If
-            None don't split (Default: None).
-        qobj_id (str): Optional, set a fixed qobj ID for all subjob qobjs.
+    num_shot_jobs, shot_mod = divmod(qobj.config.shots, max_shot_size)
+    qobj_list = []
 
-    Returns:
-        List: A list of qobjs.
-    """
+    if shot_mod == 0 and num_shot_jobs == 1:
+        return qobj
+
+    if shot_mod > 0:
+        setattr(qobj.config, "shots", shot_mod)
+        for experiment in qobj.experiments:
+            _id = str(uuid.uuid4())
+            setattr(experiment.header, "metadata", {"id": _id})
```

Comment:
Mentioned below, but is this erasing any other metadata? shouldn't you just update the field like:
```
experiment.header.metadata["id"] = _id
```